docs/src/main/asciidoc/security-openid-connect-multitenancy.adoc

```diff
@@ -844,6 +844,73 @@ The default tenant configuration is automatically disabled when `quarkus.oidc.au
 
 Note that tenant specific configurations can also be disabled, for example: `quarkus.oidc.tenant-a.tenant-enabled=false`.
 
+== Authentication outside of the HTTP request context
+
+Sometimes it is practical to be able to create `SecurityIdentity` for given token when there is no active request context.
+The `quarkus-oidc` extension uses its own `OidcIdentityProvider` to convert a token to a `SecurityIdentity` instance.
+When the provider is used during an HTTP request, the tenant configuration can be resolved as described in the <<static-tenant-resolution>> section.
+However, when there is no active HTTP request, you need to select tenant explicitly with the `io.quarkus.oidc.TenantFeature` qualifier.
+One situation when you need to select the tenant configuration this way is when you are processing messages with the xref:vertx.adoc#event-bus[Vert.x event bus].
+In the example below, the 'product-order' message is consumed within different CDI request context, therefore injected `SecurityIdentity` would be anonymous.
+
+[source,java]
+----
+package org.acme.quickstart.oidc;
+
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import org.eclipse.microprofile.jwt.JsonWebToken;
+import io.vertx.core.eventbus.EventBus;
+
+@Path("order")
+public class OrderResource {
+
+    @Inject
+    EventBus eventBus;
+
+    @Inject
+    JsonWebToken accessToken;
+
+    @RolesAllowed("customer")
+    @POST
+    public void order(String product) {
+        eventBus.publish("product-order", new Product(product, 1, accessToken.getRawToken()));
+    }
+}
+----
+
+[source,java]
+----
+package org.acme.quickstart.oidc;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import io.quarkus.oidc.TenantFeature;
+import io.quarkus.oidc.runtime.OidcIdentityProvider;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.vertx.ConsumeEvent;
+import io.smallrye.common.annotation.Blocking;
+
+@ApplicationScoped
+public class OrderService {
+
+    @TenantFeature("tenantId")
+    @Inject
+    OidcIdentityProvider identityProvider;
```

Comment:
It looks good but I'm concerned of making `OidcIdentityProvider` as the user centric API point. `import io.quarkus.oidc.runtime.*` package keeps API, classes, which I expect users interact with.

I've been thinking of a dedicated API interface `TenantIdentyProvider` for example, and its `default` implementation would call the OidcIdentityProvider directly, or may be it will get a list of all identity providers, and get those which accept token requests to verify it, or users would do something else with custom implementations.

IMHO it will give us more flexibility, what do you think ?

Reply:
@michalvavrik If we don't do it now then it may end up going directly to 3.6.0, as you may be overloaded as you already are :-).  How about:
```
public interface TenantIdentyProvider {
  Uni<SecurityIdentity> authenticate(AccessTokenCredential token);
}
```
for now ? OidcIdentityProvider makes some decision depending on whether it is an access or id token; we can add another method(s) later on depending on the requirement

Reply:
Fine with me, I'd still expect this to work fine for id token once there is use case.

Reply:
Done. Please feel free to suggest changes.